packages/gensx/tests/utils/executeWithCheckpoints.ts

```diff
@@ -0,0 +1,80 @@
+import zlib from "node:zlib";
+
+import { vi } from "vitest";
+
+import { CheckpointManager } from "@/checkpoint";
+import { ExecutionNode } from "@/checkpoint";
+import { withContext } from "@/context";
+import { ExecutionContext } from "@/context";
+import { gsx } from "@/index";
+import { ExecutableValue } from "@/types";
+import { createWorkflowContext } from "@/workflow-context";
+
+// Add types for fetch API
+export type FetchInput = Parameters<typeof fetch>[0];
+export type FetchInit = Parameters<typeof fetch>[1];
+
+/**
+ * Helper to execute a workflow with checkpoint tracking
+ * Returns both the execution result and recorded checkpoints for verification
+ */
+
+// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-parameters
+export async function executeWithCheckpoints<T>(
+  element: ExecutableValue,
+): Promise<{
+  result: T;
+  checkpoints: ExecutionNode[];
+  checkpointManager: CheckpointManager;
+}> {
+  const checkpoints: ExecutionNode[] = [];
+
+  // Set up fetch mock to capture checkpoints
+  global.fetch = vi
```

Comment:
Any worry that this sets up a moch and the caller has to know to tear it down? 

Also do we need to update tsconfig to make sure this doesn't end up in the bundle or does that happen automatically since this is in the test folder?

Reply:
Tree shaking will handle the bundle problem.

Let me add resetAllMocks to a top level afterEach and that should cover it.